contentreview-impl/impl/src/test/org/sakaiproject/contentreview/logic/TurnitinImplTest.java

```diff
@@ -30,5 +88,195 @@ public void testFileEscape() {
 		
 	}
 	
+	public void testCreateAssignment() throws Exception {
+		
+		M_util = new FakeTiiUtil();
+		M_assi = createMock(AssignmentService.class);
+		M_ss = createMock(SiteService.class);
+		M_lti = createMock(LTIService.class);
+		M_util.setLtiService(M_lti);
+
+		TurnitinReviewServiceImpl tiiService = new TurnitinReviewServiceImpl();
+
+		FakeSiteAdvisor siteAdvisor = new FakeSiteAdvisor();
+		tiiService.setSiteService(M_ss);
+		tiiService.setAssignmentService(M_assi);
+
+		Map opts = new HashMap();        
+        opts.put("submit_papers_to", "0");
+		opts.put("report_gen_speed", "1");
+        opts.put("institution_check", "0");
+        opts.put("internet_check", "0");
+        opts.put("journal_check", "0");
+        opts.put("s_paper_check", "0");
+        opts.put("s_view_report", "0");
+		opts.put("allow_any_file", "0");
+       	opts.put("exclude_biblio", "0");
+		opts.put("exclude_quoted", "0");    
+        opts.put("exclude_type", "1");
+        opts.put("exclude_value", "5");
+		opts.put("late_accept_flag", "1");
+        SimpleDateFormat dform = ((SimpleDateFormat) DateFormat.getDateInstance());
+		dform.applyPattern("yyyy-MM-dd'T'HH:mm");
+		opts.put("isostart", dform.format(new Date()));
+		opts.put("isodue", dform.format(new Date()));
+		opts.put("title", "Title");
+		opts.put("descr", "Instructions");
+		opts.put("points", 100);		
+		
+		Site siteA = new FakeSite("siteId");
```

Comment:
On 11 March 2016 at 13:41, bgarciaentornos <notifications@github.com> wrote:
>
> I took it from some other tool and thought it was nice because it , so I
> didn't have to override all the necessary methods again. Not sure about
> which one is the better practice, I can change it if you want.
>
Normally a mocking library will provide defaults for all the methods so you
only need to mock out calls that you want to be different from the default
implementation.

-- 
  Matthew Buckett, VLE Developer, IT Services, University of Oxford